system/jlib/jutil.cpp

```diff
@@ -992,6 +992,29 @@ jlib_decl int rand_r(unsigned int *seed)
 #endif
 #endif
 
+//------------------------------------------------------------------------
+//static Mersenne Twister 32 bit Pseudo Random Number Generator (PRNG) wrapper
+static class CMTRandomNumberGenerator32 : implements IMTRandomNumberGenerator32
+{
+    std::mt19937 mtPRNG;//instance of Mersenne Twister Random Number Generator
+public:
+    CMTRandomNumberGenerator32()
+    {
+        std::random_device seedGen;//seed with uniformly-distributed integer random number generator that produces non-deterministic random numbers
+        mtPRNG.seed(seedGen());
+    }
+
+    inline virtual size32_t rand32()    override    { return (size32_t)mtPRNG(); }
```

Comment:
I don't think this is thread safe.  There is no guarantee the object can be accessed from multiple threads at the same time.